### DISCUSSION TITLE
Is the project still maintained?

### DISCUSSION THREAD

**Discussion by red17electro** [facebookexperimental/Recoil#2171] [General]
Hi everyone,

I am currently considering using Recoil in production (among Zustand and Jotai). Is the library still maintained and are there plans to do so in the future?

Thanks!

**Comment by kevflynn** [comment]
agree! thought this was a great lightweight alternative to redux and would love to see it continue dev

**Reply by ayoubkhan558** [reply]
Its not lightweight, instead its bundle is size large than redux.

**Comment by FokkeZB** [comment]
@victorelu made me aware that the core contributor @drarmstr has been laid off in January, as he self shared on LinkedIn: https://www.linkedin.com/posts/douglas-armstrong-384a342_opentowork-meta-layoffs-activity-6996668167511052288-UoE4

I think that explains the radio silence, although the code frequency never has been high and mostly limited to type fixes since October: https://github.com/facebookexperimental/Recoil/graphs/code-frequency

I'm afraid we'll have to look into https://jotai.org/

**Comment by ninnepinne** [comment]
@ilya-klyuchnikov @panw @serhaty @sromku @zinuzoid
Hello Meta people.
Could you please communicate any plans for this tool/repository.
It is by far the most starred @metaexperimental project and is probably actively used by thousands of projects (mine included).
It's paradigm has even inspired other popular libraries (jotai etc).
I think it would be sad and a missed opportunity for Meta to just let this project die.

**Comment by thielium** [comment]
Maybe @davidmccabe ?

**Comment by a-sand-mp** [comment]
There was a [commit](https://github.com/facebookexperimental/Recoil/commit/a37a0318aa118e52e8bd39f2338ca653ffe8ae88) yesterday by @drarmstr, so maybe there's hope!

**Reply by FokkeZB** [reply]
Yes, it was announced on Discord that Meta has moved the source of truth to GitHub and will open Recoil up for external maintainers!

I expect we’ll hear more about that via official channels soon.

**Reply by tpatalas** [reply]
Such wonderful news

**Reply by napotopia** [reply]
> Yes, it was announced on Discord that Meta has moved the source of truth to GitHub and will open Recoil up for external maintainers!
> 
> I expect we’ll hear more about that via official channels soon.

Do you have a discord link to the announcement?

**Reply by FokkeZB** [reply]
You can invite yourself via https://discord.gg/QBKkfM7M and follow along in #general.

**Reply by tomo-local** [reply]
@FokkeZB 
Invitation link is broken.
Can I get a new invite link?

**Comment by AjaxSolutions** [comment]
This project might be dead in the water.

If you are looking for an atomic state management library, try Jotai, Nano Stores or Reatom, all actively maintained.

https://github.com/pmndrs/jotai
https://github.com/nanostores/nanostores
https://github.com/artalar/reatom

**Reply by zbyte64** [reply]
Jotai is a good choice as long as you don't have a lot of "async dependent" selectors, otherwise it requires you to laboriously mark your selector and their dependents as async. 

I had not seen nanostores before and I wish I had before we switched from Recoil to Redux because it does seem to make async dependencies rather simple in their use.

**Reply by AjaxSolutions** [reply]
@zbyte64 Do you have a small example illustrating your point?  Are you referring to Jotai V1 or V2?

I don't like Redux very well because it doesn't have a very good support for async programming.

Nano stores is recommended by the Astro team.

https://docs.astro.build/en/core-concepts/sharing-state/

Jotaii vs Nano stores:

https://github.com/pmndrs/jotai/discussions/1811

"Ecosystem: Jotai has integrations with other state libraries. It's the polyglot store framework that lets you connect redux, zustand, and valtio across a single app.
Maintainability: Jotai lets you replace a primitive atom with a derived atom, and no downstream atoms need to be updated. This allows for easier refactoring, particularly for library authors. For example, it is easy to replace atom with atomWithStorage without other app changes.
Adoption: Jotai has more downloads. That momentum means something.
On the flip side, Nanostores wins for their simple imperative API and multi framework support."

**Comment by rohamgoudarz** [comment]
**Are there any updates regarding Recoil's maintenance or leaving it to hands of external contributors?**

**Reply by FokkeZB** [reply]
I was involved in discussions about opening up to external maintainers, but that became a dead end when the last few who worked on Recoil were laid off. 😞

**Comment by listepo** [comment]
I think the project must be migrated to rescript, which will help breathe life into it. As good state management for rescript, which allows to use for js, ts, and flow.

**Comment by parweb** [comment]
so ?
